src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6591,12 +7269,32 @@ finalize_myhook(void)
 		D_MUTEX_DESTROY(&lock_mmap);
 		D_MUTEX_DESTROY(&lock_fd_dup2ed);
 
+		if (fd_255_reserved)
+			libc_close(255);
+		if (fd_dummy >= 0)
+			libc_close(fd_dummy);
+
 		if (hook_enabled_bak)
 			uninstall_hook();
 		else
 			free_memory_in_hook();
 	}
 
+	if (compatible_mode) {
+		/* Free record left in fd hash table then destroy it */
+		while (1) {
+			rlink = d_hash_rec_first(fd_hash);
+			if (rlink == NULL)
+				break;
+			d_hash_rec_decref(fd_hash, rlink);
+		}
```

Comment:
I notice this doesn't close anything, see my earlier comment about closing fds in the hash table free function, what impact would that have here?